File: solutions/go/prime-factors/1/prime_factors.go
package prime

func Factors(n int64) []int64 {
	// prime factors of a number
	var result []int64

	// loop through all numbers from 2 to n
	for i := int64(2); i <= n; i++ {
		// if n is divisible by i
		for n%i == 0 {
			// add i to result
			result = append(result, i)
			// divide n by i
			n = n / i
		}
	}
	return result
}

File: solutions/typescript/prime-factors/1/prime-factors.ts
export function calculatePrimeFactors(input: number): number[] { 
  let factors: number[] = [];
  let divisor = 2;

  if (input === 1) {
    return factors;
  }

  if (input === 2) {
    factors.push(input);
    return factors;
  }

  while (input >= 2) {
    if (input % divisor === 0) {
      factors.push(divisor);
      input = input / divisor;
    } else {
      divisor++;
    }
  }
  return factors;
}
